homeassistant/components/light/mqtt.py

```diff
@@ -444,6 +451,10 @@ def supported_features(self):
             hs_color = kwargs[ATTR_HS_COLOR]
             brightness = kwargs.get(
                 ATTR_BRIGHTNESS, self._brightness if self._brightness else 255)
+
+            if not self._scale_rgb_with_brightness:
+                brightness = 255
+
```

Comment:
brightness was just set in L452, can you work the **if** so that brightness doesn't get set twice ?